### ISSUE TITLE
"Start Over" button is unclear

### ISSUE THREAD

**Issue by KatiePro** [Microsoft/BotFramework-Emulator#630]
Pressing "Start Over" does not create a new instance of the conversation. State persists.

Documentation or additional information should be added to make this clear to developers early on.

**Comment by justinwilaby** [comment]
@KatiePro - 

Thank you for the submission.

pressing the `start over` button should [reset the state and begin a new conversation](https://github.com/Microsoft/BotFramework-Emulator/blob/v4/packages/app/client/src/ui/editor/emulator/emulator.tsx#L202). Since this is kicked off by the rendering process, you should be able to open dev tools (Ctrl + Shift + i) and look in the console for any errors. 

I've tested this locally and it all seems well and good. Was there a unique configuration you may be using or do you have multiple chat dialogs open?

**Comment by KatiePro** [comment]
I've included some code that replicates this behavior in emulator:

```javascript
// Listen for incoming requests
server.post('/api/messages', (req, res) => {

    adapter.processActivity(req, res, async (context) => {
        // const user = userState.get(context);
        const convo = convoState.get(context);
        const dc = dialogs.createContext(context, convo!);

        /*
         * This is the line of code most developers began with.
         * Using ONLY this condition (and not the nested if that directly follows),
         * pressing "Start Over" will trigger a message to be resent each time
         */
        if (context.activity.type === 'conversationUpdate' && context.activity.membersAdded![0].name !== 'Bot') {
            const user = userState.get(context);

            /*
             * This happens only once, regardless of pressing "Start Over."
             *
             * Pressing "Start Over" in this case does not actually
             * reset userState.
             */
            if (!user!.welcomed) {
                user!.welcomed = true;
                await context.sendActivity(`Hey there! I'm a bot welcoming you once!`);
            }

            /*
             * Again, this does and should reappear each time after pressing "Start Over"
             * because it's not within the nested if directly above.
             */
            await showMenu(context);
        }
```

The console did not output any errors for me. I downloaded the v4.0.15-alpha emulator using botframework-emulator-setup-4.0.15-alpha.exe.

I recently took part in a hack where this code produced the same output for each team member. They had the same version of the emulator.

**Comment by vishwacsena** [comment]
Conversation and user states are two different concepts. User state stays alive until there is an actual user talking to the bot. Although start over resets the conversation ID, it has no bearing on the user ID. In fact for emulator, the user id is always the same and you should be able to verify this by having your bot running, talk to it in emulator, restart emulator and your messages will still look like the same user.

**Comment by vishwacsena** [comment]
Closing due to lack of activity. Please reactivate if you need additional clarity.

**Comment by jorn1** [comment]
Please fix this.

**Comment by vishwacsena** [comment]
@jorn1, the latest emulator release (4.2) does address this with two changes. 

- restart Conversation now resets both Conversation and user ID 
- Optionally you can also restart a conversation and keep the same user ID

**Comment by jorn1** [comment]
But if the Conversation ID changes then it's not the same conversation right? Or is the conversation ID not the key to the conversation? why call it restart conversation if it does not restart the conversation but creates a new one???

If i tell you to restart your computer, do you shut down your computer and get another computer?? Nobody would understand it in that way. 

This setup would make more sense:
<button>connect to bot<button>
<botton>new userID<button>
<button>new ConversationID><button>

something like that.
or maybe make the last two buttons checkboxes instead.

**Comment by vishwacsena** [comment]
Reopening for feature crew triage.

@mewa1024: @tonyanziano

**Comment by tonyanziano** [comment]
Closed as 'Won't Fix' for now. Discussion in #1171
